src/main/java/com/zendesk/maxwell/MaxwellMysqlStatus.java

```diff
@@ -15,13 +15,17 @@ private String sqlStatement(String variableName) {
 		return "SHOW VARIABLES LIKE '" + variableName + "'";
 	}
 
-	private String getVariableState(String variableName) throws SQLException, MaxwellCompatibilityError {
+	private String getVariableState(String variableName, boolean throwOnMissing) throws SQLException, MaxwellCompatibilityError {
```

Comment:
can we capture this into ensureVariableState("read_only", "OFF") and getVariableState("binlog_format") ? Lots of code duplication here.